### PR TITLE
Use camelCase wherever possible

### LINKED ISSUE #192: Find all places where `serde` can do automatic variant/field renaming
## Description

We should look into finding all places that could benefit from automatically renaming struct fields and enum variants. 

## Are you planning to do it yourself in a pull request?

Yes.

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Rather than individual renaming camelCase fields, this PR uses serde `rename_all = "camelCase"` wherever possible.

## Links to any relevant issues

- Closes #192 

## Type of change

Choose a type of change, and delete any options that are not relevant.

- Enhancement (a non-breaking change which adds functionality)

## How the change has been tested

Ran local tests to ensure renaming was correct, particularly with struct enum variants.

## Change checklist

Tick the boxes that are relevant to your changes, and delete any items that are not.

- [ ] I have followed the contribution guidelines for this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that new and existing unit tests pass locally with my changes

### CODE REVIEW

File: bindings/core/src/method/client.rs [hunk 1]
```diff
@@ -153,34 +139,33 @@ pub enum ClientMethod {
...
         /// Prepared transaction data
-        #[serde(rename = "preparedTransactionData")]
         prepared_transaction_data: PreparedTransactionDataDto,
     },
     /// Build a block containing the specified payload and post it to the network.
+    #[serde(rename_all = "camelCase")]
     PostBlockPayload {
         /// The payload to send
-        #[serde(rename = "payload")]
         payload_dto: PayloadDto,
```
Comment (Thoralf-M) [inline bindings/core/src/method/client.rs]: Renaming to payload was good, otherwise it's breaking?

Comment (DaughterOfMars) [reply bindings/core/src/method/client.rs]: How annoying. Do we prefer unifying the naming or avoiding a breaking change?

Comment (Thoralf-M) [reply bindings/core/src/method/client.rs]: Bindings build this and for them it's not a Dto, but just the normal payload, so I would keep it non breaking

**Review by Thoralf-M — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

File: bindings/core/src/method/account.rs [hunk 2]
```diff
@@ -240,22 +221,22 @@ pub enum AccountMethod {
...
-        #[serde(rename = "addressesNativeTokens")]
         addresses_native_tokens: Vec<AddressNativeTokens>,
         options: Option<TransactionOptionsDto>,
     },
     /// Send nft.
     /// Expected response: [`SentTransaction`](crate::Response::SentTransaction)
+    #[serde(rename_all = "camelCase")]
     SendNft {
-        #[serde(rename = "addressesAndNftIds")]
         addresses_nft_ids: Vec<AddressAndNftId>,
```
Comment (thibault-martinez) [inline bindings/core/src/method/account.rs]: This rename was not simply camelCase, but it's the message interface so I don't think it matters. @Thoralf-M can you confirm?

Comment (thibault-martinez) [reply bindings/core/src/method/account.rs]: Maybe we just rename the field?

Comment (DaughterOfMars) [reply bindings/core/src/method/account.rs]: Can do

File: sdk/src/wallet/account/operations/transaction/prepare_output.rs [hunk 3]
```diff
@@ -404,10 +404,9 @@ pub struct OutputOptions {
 }
 
 #[derive(Debug, Default, Clone, Eq, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
 pub struct Assets {
-    #[serde(rename = "nativeToken")]
     pub native_tokens: Option<Vec<NativeToken>>,
```
Comment (thibault-martinez) [inline sdk/src/wallet/account/operations/transaction/prepare_output.rs:409]: There is a mismatch here. @Thoralf-M do we ever store this? I don't think so?

Comment (DaughterOfMars) [reply sdk/src/wallet/account/operations/transaction/prepare_output.rs:409]: Luckily the Dto uses plural, so this does not affect anything.

File: sdk/src/client/message_interface/message.rs [hunk 4]
```diff
@@ -213,18 +197,17 @@ pub enum Message {
...
         secret_manager: SecretManagerDto,
         /// Mnemonic
         mnemonic: String,
     },
     /// Build a block containing the specified payload and post it to the network.
     PostBlockPayload {
         /// The payload to send
-        #[serde(rename = "payload")]
-        payload_dto: PayloadDto,
+        payload: PayloadDto,
```
Comment (thibault-martinez) [inline sdk/src/client/message_interface/message.rs:210]: This deserves a CHANGELOG entry

File: bindings/core/src/method_handler/account.rs [hunk 5]
```diff
@@ -300,24 +300,24 @@ pub(crate) async fn call_account_method_internal(account: &AccountHandle, method
...
-                    addresses_native_tokens.clone(),
+                    addresses_and_native_tokens.clone(),
                     options.as_ref().map(TransactionOptions::try_from_dto).transpose()?,
                 )
                 .await?;
             Response::SentTransaction(TransactionDto::from(&transaction))
         }
         AccountMethod::SendNft {
-            addresses_nft_ids,
+            addresses_and_nft_ids,
```
Comment (thibault-martinez) [inline bindings/core/src/method_handler/account.rs:315]: These 2 as well in the changelog

File: sdk/src/wallet/bindings/nodejs/lib/Account.ts [hunk 6]
```diff
@@ -926,21 +926,21 @@ export class Account {
 
     /**
      * Send native tokens.
-     * @param addressesNativeTokens Addresses amounts and native tokens.
+     * @param addressesAndNativeTokens Addresses amounts and native tokens.
      * @param transactionOptions The options to define a `RemainderValueStrategy`
      * or custom inputs.
      * @returns The sent transaction.
      */
     async sendNativeTokens(
```
Comment (thibault-martinez) [inline sdk/src/wallet/bindings/nodejs/lib/Account.ts:934]: Does `SendNft` not need a change as well?

Comment (DaughterOfMars) [reply sdk/src/wallet/bindings/nodejs/lib/Account.ts:934]: It already was named with `And`, so I don't think so

**Review by thibault-martinez — APPROVED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**